lua/ide/components/bufferlist/component.lua

```diff
@@ -133,7 +144,19 @@ BufferListComponent.new = function(name, config)
     end
 
     function self.refresh()
-        local bufs = vim.tbl_map(function(buf)
+        local cur_buf = vim.api.nvim_get_current_buf()
+        local listed_bufs = libbuf.get_listed_bufs()
+        if #listed_bufs == 0 then
+            return
+        end
+
+        self.bufs = {}
+        if self.config.current_buffer_top then
+            -- reserve first index for current buffer.
+            self.bufs = {""}
```

Comment:
why empty string? can't we just use the `sort` module and sort the current buffer to the top? I don't think it's a great idea to mix types here, bufs is a list of tables with some info about each buffer.

What I'd recommend instead is just updating `refresh` to also include `is_current = true/false` and then sort on that key.

Reply:
I think we can avoid a search by just placing a sentinel in the first index, which seems a bit quicker. We only enter these functions when we are inside a valid source code buffer, so we'll always have an "is_current" and a buffer in index 1.

Reply:
True, but I personally think in this case it's worth doing the extra bit of work to sort rather than use a sentinel value because it's much easier to read/understand/reason about, and the sorting is really not that much extra work. Lua is extremely fast and we aren't sorting many items, most likely 10-50 if I had to estimate how many buffers people have open at once.